packages/apps/reputation-oracle/server/src/modules/nda/nda.service.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { NDAService } from './nda.service';
+import { UserRepository } from '../user/user.repository';
+import { AuthConfigService } from '../../config/auth-config.service';
+import { NDASignatureDto } from './nda.dto';
+import { NDAError, NDAErrorMessage } from './nda.error';
+import { faker } from '@faker-js/faker/.';
+
+const mockUserRepository = {
+  updateOne: jest.fn(),
+};
+const validNdaUrl = faker.internet.url();
+const mockAuthConfigService = {
+  latestNdaUrl: validNdaUrl,
+};
+
+describe('NDAService', () => {
+  let service: NDAService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        NDAService,
+        { provide: UserRepository, useValue: mockUserRepository },
+        { provide: AuthConfigService, useValue: mockAuthConfigService },
+      ],
+    }).compile();
+
+    service = module.get<NDAService>(NDAService);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('signNDA', () => {
+    const user: any = {
+      id: 1,
+      email: faker.internet.email(),
+      password: 'password',
```

Comment:
Let's remove `password` since it's not used.

Also please use faker to generate `id` and avoid using `any`, you can use `Pick<UserEntity, 'id' | 'email' | 'ndaSigned`

Ideally we just implement `generateUser` in `test/fixtures`, but you can leave this part for refactoring phase, where having `UserEntity` type usage will help us to find all places that need updates